core/templates/domain/editor/undo_redo/ChangeObjectFactory.ts

```diff
@@ -26,53 +26,13 @@ import cloneDeep from 'lodash/cloneDeep';
 import { downgradeInjectable } from '@angular/upgrade/static';
 import { Injectable } from '@angular/core';
 
-import { MisconceptionBackendDict } from
-  'domain/skill/MisconceptionObjectFactory';
-
-interface BackendChangeObject {
-  // This interface may not have all the possible values.
-  // Add properties with a '?' if you find any that needs to be here.
-  'cmd'?: string;
-  'property_name'?: string;
-
-  'state_name'?: string;
-  'old_state_name'?: string;
-  'new_state_name'?: string;
-
-  'new_value'?: Object;
-  'old_value'?: Object;
-
-  'exploration_id'?: string;
-
-  'new_misconception_dict'?: MisconceptionBackendDict;
-  'misconception_id'?: string;
-
-  'skill_id'?: string;
-  'uncategorized_skill_id'?: string;
-  'new_uncategorized_skill_id'?: string;
-  'skill_difficulty'?: number;
-
-  'explanations'?: string[];
-  'difficulty'?: string;
-
-  'node_id'?: string;
-
-  'story_id'?: string;
-
-  'subtopic_id'?: number;
-  'new_subtopic_id'?: number;
-  'old_subtopic_id'?: number;
-
-  'version_number'?: number;
-}
-
 export class Change {
-  _backendChangeObject: BackendChangeObject;
+  _backendChangeObject: Object;
```

Comment:
I added a generic type `Object` here because I realized that adding different kinds of changes we use don't make sense here because this service that technically accept any kind of change.

Reply:
But then we should handle it similarly to playthroughs / interactions and fully enumerate those kinds of changes, otherwise we're going against the principles of type safety. (One day, we might be able to tighten this up yet further, but I think "Object" is too loose.)

Reply:
Makes sense. Will add types here.